federation-js/src/composition/utils.ts

```diff
@@ -480,7 +508,7 @@ export function diffTypeNodes(
   return {
     name: typeNameDiff,
     kind: kindDiff,
-    fields: fieldsDiff,
+    fields: { ...fieldsDiff, ...inputValuesDiff },
```

Comment:
I think we want to break this out into its own key rather than combine with fields when we return.

So we might instead return:
```js
{
  // ...,
  fields: fieldsDiff,
  inputValues: inputValuesDiff,
  // ...,
}
```

When we combine the two objects like this, we still have a potential naming collision. You can imagine the case where we had a real `fieldsDiff` and no `inputValuesDiff`, but they have a name in common (similar to the case you encountered). We could have these two diffs:

```js
// fieldsDiff
{ identifier: ['String', 'String!'] }

// inputValuesDiff
{ identifier: [] } 
```

And now the resulting diff we report is:
```js
{ identifier: [] }
```
...due to the "overwriting" nature of spread and the way we've overloaded the `fields` key here.

When you make this change, please take a look at the `typeNodesAreEquivalent` function below. I'd like to see your new `inputValues` object receive a similar treatment to the `fields` object within that function.